### PR TITLE
[Issue 3837][pulsar-client-tools]Delete schema when deleting a topic

### PR DESCRIPTION
**Pull request by tuteng**
Fixes: #3837 

### Motivation

* Providing an option to delete schema when deleting a topic.

### Modifications

* Add deleteSchema parameter for delete and delete-partitioned-topic

### Verifying this change

Test pass

### CODE REVIEW

**Comment by tuteng** [comment]
> @tuteng
> 
> overall looks good. can you try to add a test for it?
> 
> you can check `pulsar-client-tools-test` to see how we test `CmdFunctions`.

I did not find the relevant test in `CmdFunctions`, but i added the relevant tests in `PulsarAdminToolTest`. Please review the code again, thanks. @sijie

File: pulsar-client-tools-test/src/test/java/org/apache/pulsar/admin/cli/PulsarAdminToolTest.java [hunk 1]
```diff
@@ -558,10 +559,12 @@ void topics() throws Exception {
...
         Topics mockTopics = mock(Topics.class);
         when(admin.topics()).thenReturn(mockTopics);
+        Schemas mockSchemas = mock(Schemas.class);
+        when(admin.schemas()).thenReturn(mockSchemas);
 
         CmdTopics cmdTopics = new CmdTopics(admin);
 
-        cmdTopics.run(split("delete persistent://myprop/clust/ns1/ds1"));
+        cmdTopics.run(split("delete persistent://myprop/clust/ns1/ds1 -d"));
         verify(mockTopics).delete("persistent://myprop/clust/ns1/ds1", false);
```
Comment (sijie) [inline pulsar-client-tools-test/src/test/java/org/apache/pulsar/admin/cli/PulsarAdminToolTest.java]: can you verify mockSchemas called deleteSchema once?

**Comment by tuteng** [comment]
run java8 tests

**Comment by tuteng** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by tuteng** [comment]
run java8 tests

**Pull request merged**